src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const parentElement = document.querySelector('ul');
+const employees = Array.from(document.querySelectorAll('li'));
+
+function getSalary(employee) {
+  const salaryString = employee.getAttribute('data-salary');
+  const cleanSalary = salaryString.replace(/[^0-9.-]+/g, '');
+
+  return parseInt(cleanSalary, 10);
+}
+
+function sortList() {
+  employees.sort((a, b) => getSalary(b) - getSalary(a));
+
+  parentElement.innerHTML = '';
```

Comment:
Clearing the innerHTML of 'parentElement' removes all child nodes. Ensure that this is the intended behavior and that there are no other elements within 'ul' that should be preserved.